rotkehlchen/chain/evm/decoding/uniswap/v3/utils.py

```diff
@@ -123,6 +125,8 @@ def get_uniswap_v3_position_price(
             abi=uniswap_v3_nft_manager.abi,
             method_name='positions',
             arguments=[int(collectible_id)],
+            block_identifier=block_identifier,
+            call_order=evm_inquirer.archive_nodes_call_order(),
```

Comment:
I believe you want something smarter. If it is latest you don't want extra logic on top of the order and only when it isn't latest then query with archive nodes. Moreover when making historical queries you shouldn't use non-archive nodes